File: TSL UHF ASCII 2 SDK DotNet v1.1/ASCII Protocol Switch Sample/AsciiSwitchSample/Service.cs
//-----------------------------------------------------------------------
// <copyright file="Service.cs" company="Technology Solutions UK Ltd"> 
//     Copyright (c) 2013 Technology Solutions UK Ltd. All rights reserved. 
// </copyright> 
// <author>Robin Stone</author>
//-----------------------------------------------------------------------
namespace TechnologySolutions.AsciiProtocol.Sample
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    using Services;
    using TechnologySolutions.Rfid.AsciiProtocol;

    /// <summary>
    /// Provides access to application services
    /// </summary>
    public class Service
    {
        /// <summary>
        /// Gets or sets a responder that copies messages to the display
        /// </summary>
        public static DisplayResponder DisplayResponder { get; set; }

        /// <summary>
        /// Gets or sets the reader service
        /// </summary>
        public static ReaderService Reader { get; set; }

        /// <summary>
        /// Gets the application settings
        /// </summary>
        internal static Properties.Settings Settings
        {
            get
            {
                return Properties.Settings.Default;
            }
        }

        /// <summary>
        /// Sets up the application services
        /// </summary>
        public static void StartUp()
        {
            DisplayResponder = new DisplayResponder();
            Reader = new ReaderService();
        }

        /// <summary>
        /// Shuts down the application services
        /// </summary>
        public static void ShutDown()
        {
            Dispose(Reader);
            Dispose(DisplayResponder);
        }

        /// <summary>
        /// Disposes a service if the service implements <see cref="IDisposable"/>
        /// </summary>
        /// <param name="service">The service to dispose</param>
        /// <returns>True if the service was disposed</returns>
        public static bool Dispose(object service)
        {
            IDisposable disposable;

            disposable = service as IDisposable;
            if (disposable != null)
            {
                disposable.Dispose();
                return true;
            }

            return false;
        }
    }
}

File: TSL UHF ASCII 2 SDK DotNet v1.1/ASCII Protocol Inventory Sample VB/AsciiProtocolInventoryFx.ConvertedToVBNet/Service.vb

Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Imports TechnologySolutions.Rfid.AsciiProtocol

Public Class Service
	Public Shared Sub StartUp()
		DisplayResponder = New DisplayResponder()
		Reader = New ReaderService()
	End Sub

	Public Shared Property Reader() As ReaderService
		Get
			Return m_Reader
		End Get
		Set
			m_Reader = Value
		End Set
	End Property
	Private Shared m_Reader As ReaderService

	Public Shared Property DisplayResponder() As DisplayResponder
		Get
			Return m_DisplayResponder
		End Get
		Set
			m_DisplayResponder = Value
		End Set
	End Property
	Private Shared m_DisplayResponder As DisplayResponder

	Friend Shared ReadOnly Property Settings() As Properties.Settings
		Get
			Return Properties.Settings.[Default]
		End Get
	End Property

	Public Shared Sub ShutDown()
		Dispose(Reader)
		Dispose(DisplayResponder)
	End Sub

	Public Shared Function Dispose(value As Object) As Boolean
		Dim disposable As IDisposable

		disposable = TryCast(value, IDisposable)
		If disposable IsNot Nothing Then
			disposable.Dispose()
			Return True
		End If

		Return False
	End Function
End Class
